projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts

```diff
@@ -4344,7 +4353,195 @@ describe('IgxGrid - Filtering actions - Excel style filtering #grid', () => {
+        tick(100);
+        fix.detectChanges();
+
+        ControlsFunction.verifyButtonIsDisabled(applyButton);
+        }));
+
+        it('Should add list items to current filtered items when "Add to current filter selection" is selected.', fakeAsync(() => {
+        const totalListItems = [];
+
+        // Open excel style custom filtering dialog.
+        GridFunctions.clickExcelFilterIconFromCode(fix, grid, 'Downloads');
+
+        // Type string in search box.
+        let searchComponent = GridFunctions.getExcelStyleSearchComponent(fix);
+        let inputNativeElement = GridFunctions.getExcelStyleSearchComponentInput(fix, searchComponent);
+        UIInteractions.clickAndSendInputElementValue(inputNativeElement, '5', fix);
+        tick(100);
+        fix.detectChanges();
+
+        let listItems = GridFunctions.getExcelStyleSearchComponentListItems(fix, searchComponent)
+            .splice(2)
+            .map(c => c.innerText);
+
+        listItems.forEach(c => totalListItems.push(c));
+
+        // Click 'apply' button to apply filter.
+        GridFunctions.clickApplyExcelStyleFiltering(fix);
+        tick(100);
+        fix.detectChanges();
+
+        // Get the results and verify that they match the list items.
+        let gridCellValues = GridFunctions.getColumnCells(fix, 'Downloads')
+            .map(c => c.nativeElement.innerText);
+
+        expect(gridCellValues.length).toEqual(1);
+        expect(gridCellValues).toEqual(totalListItems);
+
+        // Open excel style custom filtering dialog again.
+        GridFunctions.clickExcelFilterIconFromCode(fix, grid, 'Downloads');
+
+        // Type string in search box.
+        searchComponent = GridFunctions.getExcelStyleSearchComponent(fix);
+        inputNativeElement = GridFunctions.getExcelStyleSearchComponentInput(fix, searchComponent);
+        UIInteractions.clickAndSendInputElementValue(inputNativeElement, '7', fix);
+        tick(100);
+        fix.detectChanges();
+
+        listItems = GridFunctions.getExcelStyleSearchComponentListItems(fix, searchComponent)
+            .splice(2)
+            .map(c => c.innerText);
+
+        listItems.forEach(c => totalListItems.push(c));
+        totalListItems.sort();
+
+        // Select 'Add to current filter selection'.
+        const checkbox = GridFunctions.getExcelStyleFilteringCheckboxes(fix);
+        checkbox[1].click();
+        tick();
+        fix.detectChanges();
+
+        // Click 'apply' button to apply filter.
+        GridFunctions.clickApplyExcelStyleFiltering(fix);
+        tick(100);
+        fix.detectChanges();
+
+        // Get the results and verify that they match the list items.
+        gridCellValues = GridFunctions.getColumnCells(fix, 'Downloads')
+            .map(c => c.nativeElement.innerText)
+            .sort();
+
+        expect(gridCellValues.length).toEqual(3);
+        expect(gridCellValues).toEqual(totalListItems);
```

Comment:
If possible - at the end of the test open the filter dialog again and verify the selection state of the 'Select All' button.